linux/mcompile/keymap/mcompile.cpp

```diff
@@ -36,65 +36,58 @@
 
 KMX_BOOL KMX_DoConvert(LPKMX_KEYBOARD kbd, KMX_BOOL bDeadkeyConversion, gint argc, gchar *argv[]);
 
-bool KMX_ImportRules( LPKMX_KEYBOARD kp,vec_dword_3D& all_vector, GdkKeymap **keymap,std::vector<KMX_DeadkeyMapping> *KMX_FDeadkeys, KMX_BOOL bDeadkeyConversion); // I4353   // I4327
+bool KMX_ImportRules(LPKMX_KEYBOARD kp, vec_dword_3D& all_vector, GdkKeymap** keymap, std::vector<KMX_DeadkeyMapping>* KMX_FDeadkeys, KMX_BOOL bDeadkeyConversion); // I4353   // I4327
+
+int run(int argc, char* argv[], char* argv_gdk[] = NULL);
+
+int KMX_GetDeadkeys(vec_dword_2D& dk_Table, KMX_WORD DeadKey, KMX_WORD* OutputPairs, GdkKeymap* keymap);
 
 std::vector<KMX_DeadkeyMapping> KMX_FDeadkeys; // I4353
 
 // Note: max is not a standard c api function or macro
 #ifndef max
-#define max(a,b)            (((a) > (b)) ? (a) : (b))
+#define max(a, b)            (((a) > (b)) ? (a) : (b))
 #endif
 
-#define _countof(a) (sizeof(a)/sizeof(*(a)))
+#define _countof(a) (sizeof(a) / sizeof(*(a)))
 
 #if defined(_WIN32) || defined(_WIN64)
   int wmain(int argc, wchar_t* argv[]) {
-    std::vector<std::u16string> str_argv_16 = convert_argvW_to_Vector_u16str( argc, argv);
-    run(argc, str_argv_16);
+    run(argc, argv);
 
 #else  // LINUX
   int main(int argc, char* argv[]) {
-    std::vector<std::u16string> str_argv_16 = convert_argv_to_Vector_u16str(argc, argv);
-    run(argc, str_argv_16, argv);
+    run(argc, argv, argv);
 #endif
-
 }
 
-int run(int argc, std::vector<std::u16string> str_argv, char* argv_ch[] = NULL){
+int run(int argc, char* argv[], char* argv_gdk[]) {
```

Comment:
yes, it would be called from Windows, but Windows doesn't use `argv_gdk`, and on Linux it's used in a `#ifdef LINUX` block, so I think it's safe to remove it. In fact, that would make the method signature identical to the one found in `windows/src/engine/mcompile/mcompile.cpp`.

Reply:
ok done